tests/app/shared/issue/assignee/assignee.component.spec.ts

```diff
@@ -0,0 +1,138 @@
+      id: 'F09-2',
+      teamMembers: [testStudent]
+  });
+
+  const dummyIssue: Issue =  Issue.createPhaseTeamResponseIssue(ISSUE_WITH_EMPTY_DESCRIPTION, dummyTeam);
+
+  const userService: UserService = new UserService(null, null);
+  userService.currentUser = testStudent;
+  const phaseService: PhaseService = new PhaseService(null, null, null, userService, null);
+  phaseService.currentPhase = Phase.phaseTeamResponse;
+  const issueService = jasmine.createSpyObj('IssueService', ['getLatestIssue', 'updateIssue']);
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        AssigneeComponent
+      ],
+      providers: [
+        UserService, IssueService, ErrorHandlingService, PhaseService, PermissionService
+      ],
+      imports: [
+        FormsModule, MaterialModule, ApolloTestingModule, HttpClientModule, BrowserAnimationsModule
+      ]
+    })
+    .overrideProvider(UserService, { useValue: userService })
+    .overrideProvider(PhaseService, { useValue: phaseService })
+    .overrideProvider(IssueService, { useValue: issueService })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(AssigneeComponent);
+    component = fixture.componentInstance;
+
+    component.team = dummyTeam;
+    component.issue = dummyIssue;
+    fixture.detectChanges();
+    debugElement = fixture.debugElement;
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should have a placeholder value of - given no assignees', () => {
+    const matPlaceholderValue: HTMLElement = debugElement.query(By.css('.mat-select-placeholder')).nativeElement;
+    expect(matPlaceholderValue).toBeDefined();
+    expect(matPlaceholderValue.innerText).toEqual('-'); // Placeholder Value
+  });
+
+  it('should be able to open the assignee selector', () => {
+    const matSelect: HTMLElement = debugElement.query(By.css('.mat-select-trigger')).nativeElement;
+
+    // Open the assignee selector
+    matSelect.click();
+    fixture.detectChanges();
+    const matOption: HTMLElement = debugElement.query(By.css('.mat-option')).nativeElement;
+    const inputElement: HTMLElement = debugElement.query(By.css('.mat-select-panel')).nativeElement;
+
+    expect(inputElement.children.length).toBe(dummyTeam.teamMembers.length);
+    expect(matOption.attributes.getNamedItem('aria-selected').value).toEqual('false');
+  });
+
+  it('should emit the issueUpdated event upon closing the MatSelect', () => {
+    spyOn(component.issueUpdated, 'emit');
+    const matSelect: HTMLElement = debugElement.query(By.css('.mat-select-trigger')).nativeElement;
+
+    // Open the assignee selector
+    matSelect.click();
+    fixture.detectChanges();
+
+    // Close the matSelect element and make relevant fake calls.
+    const matSelectElement = debugElement.query(By.css('.mat-select')).nativeElement;
+    issueService.updateIssue.and.callFake(() => of(dummyIssue));
```

Comment:
For this test, let's also perform the `mat-option` selection (i.e. choose the assignee) **before** we close the `matSelectElement`.

This will allow us to test whether the `updateAssignee()` method grabs the selected assignees from the `mat-select` and updates the issue, before emitting it.

We should return the updated issue, and not the initial dummy issue, since the updated issue will have the modified assignees (as set within the `updateAssignee()` method)
```suggestion
    issueService.updateIssue.and.callFake(updatedIssue => of(updatedIssue));
```

Reply:
Ah alright, that is understood.